### PR TITLE
Add additional rego templates

### PR DESCRIPTION
**Pull request by furi0us333**
- Adding several additional templates

### CODE REVIEW

**Comment by furi0us333** [comment]
CC @louislang @janasheehan @kylewillmon since you seemed curious

File: install_code.rego [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+   data.issue.tag == "IM0042"
+}
+
+issue contains "Package contains code execution on install" if {
+   data.issue.tag == "IM0043"
+}
+
+issue contains "Package contains code execution on install" if {
+   data.issue.tag == "IM0044"
+}
```
Comment (kylewillmon) [inline install_code.rego]:
```suggestion
# Returns a violation if there is code execution on package install
issue contains "Package contains code execution on install" if {
   data.issue.tag in {"IM0042", "IM0043", "IM0044"}
}
```

File: install_code_suspicious.rego [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import future.keywords.if
+import future.keywords.in
+
+
+# METADATA
+# scope: rule
+# schemas:
+#   - data.issue: schema.issue
+
+# Returns a violation if there is suspicious code execution on pacakge install
```
Comment (kylewillmon) [inline install_code_suspicious.rego]:
```suggestion
# Returns a violation if there is suspicious code execution on package install

```

File: malware.rego
Comment (kylewillmon) [inline malware.rego:1]: Based on the name of this file, I expected the rule to simply check if there are any malware issues. But this picks out specific tags. Is there a reason not to include all malware issues (or at least those of a certain level)?

File: obfuscated_code.rego [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+# METADATA
+# scope: rule
+# schemas:
+#   - data.issue: schema.issue
+
+# Returns a violation if the package contains obfuscated code
+issue contains "Package contains obfuscated code" if {
+   obfuscated_code
+}
```
Comment (kylewillmon) [inline obfuscated_code.rego]:
```suggestion
# METADATA
# scope: rule
# schemas:
#   - data.issue: schema.issue

# Returns a violation if the package contains obfuscated code
issue contains "Package contains obfuscated code" if {
   data.issue.tag in {"HM0029", "HM0099", "HM0023", "IM0040", "IM0041"}
}
```

**Review by louislang — APPROVED** [review]
Looks reasonable to me!

File: typosquat.rego [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import data.phylum.domain
+import data.phylum.level
+
+import future.keywords.contains
+import future.keywords.if
+
+# Returns `true` if the given dependency has a typosquat issue
+has_typosquat {
+    issues := data.dependency.issues
+    count([tag | issues[i].tag == "HM0008"; tag := issues[i].tag]) == 1
```
Comment (kylewillmon) [inline typosquat.rego]: There is no need for `count` here. Unless a dependency with 2 typosquat issues is somehow better than only having 1.

```suggestion
    some i
    data.dependency.issues[i].tag == "HM0008"
```

**Review by kylewillmon — COMMENTED** [review]
We'll probably need input from @matt-phylum here. I think these will need to be written with policy.v1 in order to be combined as intended.

File: install_code.rego (hunk 1 above)
Comment (furi0us333) [reply install_code.rego]: I figured there was a better way to do this...

File: malware.rego
Comment (furi0us333) [reply malware.rego:1]: The issue tags in this rego are essentially the "no false positive" issues. Perhaps a filename change to `confirmed_malicious.rego` or something?

File: typosquat.rego (hunk 4 above)
Comment (furi0us333) [reply typosquat.rego]: Not even sure that is possible come to think of it 🤔

**Comment by furi0us333** [comment]
I do expect there might be a need to re-write some of these, but I figured it was worth getting them out there now. At a minimum, I think we won't need the `future` imports with `v1`.

**Comment by matt-phylum** [comment]
The future imports are part of OPA. With the updated version of OPA, you should be able to replace them all with a single `import rego.v1`.

Execution-wise, it should be possible to combine these together even though they're `policy` instead of `policy.v1`. `policy.v1` can have better performance and is normally nicer to work with, and there are some features that were planned for `policy.v1` which don't make sense for `policy` policies. `policy` policies actually execute in a wrapper that makes them execute the same as `policy.v1` policies.

However, one thing that `policy.v1` added was useful data for introspection so we can display them nicely in a UI. It should work if you put the same top metadata block at the top of a `policy` policy. For a `policy.v1` policy we could theoretically give a list of the rules contained within the policy, which is difficult to do accurately and meaningfully for `policy` policies, but we don't currently do this. The top metadata block is extracted, but not yet surfaced anywhere. These properties [come from OPA](https://www.openpolicyagent.org/docs/latest/policy-language/#annotations) but we can omit any of these and add any others we want under `custom`.

For example, if you wanted to rewrite `data_exfiltration.rego` it would look something like this:

```rego
# METADATA
# title: Data exfiltration
# description: |
#   Blocks packages that may send data from your development or build environment to an outside server.
# organizations:
#   - Phylum
# schemas:
#   - data.issues: schema.issues
package policy.v1

import rego.v1

# METADATA
# title: Package contains environment variable enumeration
deny contains issue if {
    some issue in data.issues
    issue.tag == "HM0025"
}

# METADATA
# title: Package contains webhook exfiltration
deny contains issue if {
    some issue in data.issues
    issue.tag == "HM0036"
}
```

And then in the UI we can display a nice list with additional description.

**Pull request merged**